### ISSUE TITLE
Button Group attribute string option request

### ISSUE THREAD

**Issue by druckerk** [carbon-design-system/carbon-for-ibm-dotcom#9687]
### The problem

The Button Group component automatically defaults to the "tertiary" button in the first position when the number of buttons is set to two. The button shows properly in AEM when the number of buttons is set to one.  

https://www.ibm.com/standards/carbon/web-components/?path=/story/components-button-group--default

### The solution

Is it possible to add an attribute option to the button group tag where we can set which button is positioned first when the number of buttons is set to two?  For example, an attribute option like ```button-order="tertiary"```;  ```"tertiary"``` can be set as the default value.  

**Default:**
```
<dds-button-group button-order="tertiary">
          <dds-button-cta href="https://example.com" kind="tertiary">
                   Button 1
         </dds-button-cta>
          <dds-button-cta href="https://example.com" kind="primary">
                  Button 2
          </dds-button-cta>
</dds-button-group>
```

**Primary Button in First Position:**
```
<dds-button-group button-order="primary">
          <dds-button-cta href="https://example.com" kind="primary">
                   Button 1
         </dds-button-cta>
          <dds-button-cta href="https://example.com" kind="tertiary">
                  Button 2
          </dds-button-cta>
</dds-button-group>
```

### Application/website

AEM

### Business priority

Medium Priority = upcoming release but is not pressing

### What time frame would this ideally be needed by (if applicable)

If this update can be part of the next release, that would be greatly appreciated so we can update the button group to use carbon tags. Thanks.

### Examples

This is a request from the inferences team for AEM. 
Ticket: https://jsw.ibm.com/browse/ADCMS-2714

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

**Comment by RichKummer** [comment]
Hey @druckerk! The design team is currently doing some research around the `lead space` and looking into switching the order of the buttons globally - so the new ordering would be your requested primary button on the left followed by the tertiary button. 

This would be a global change to the button ordering, so the ordering would probably not be configurable. This way the buttons appear consistent across the components they are in.

**Comment by oliviaflory** [comment]
Hi @druckerk we are updating the button order and this issue should be resolved with that update #9831 and will be coming out in our next release.
